### PR TITLE
feat(ai-infra): add robotics foundation models on EKS

### PR DESCRIPTION
**Pull request by yoosful**
## Summary

Adds **[Building Robotics Foundation Models on Amazon EKS](ai-infra/robotics-foundation-models-on-eks/README.md)** under `ai-infra/`.

This sample provides an AWS deployment path for NVIDIA OSMO on Amazon EKS, plus validated robotics foundation model workflows for smoke tests, policy training, simulation, world models, EFA, and managed observability.

### Architecture Overview

image

### AWS Infrastructure

- Terraform-based EKS baseline with private networking.
- AWS managed backing services for OSMO, including Amazon RDS PostgreSQL, Amazon ElastiCache for Redis, Amazon S3, Amazon ECR, AWS KMS, and IRSA.
- Karpenter GPU node provisioning for the validated G7e and G6e workflow paths.
- Optional HTTPS admin ingress and optional AMP/AMG observability setup.

### Kubernetes And OSMO Deployment

- Deployment wrappers for Karpenter, KAI Scheduler, NVIDIA GPU Operator, AWS EFA device plugin, and OSMO.
- CPU and GPU smoke workflows to validate the cluster, OSMO backend, KAI scheduling, Karpenter provisioning, and GPU runtime path.

### Validated Robotics Workflows

- GR00T SO100 fine-tuning, including a two-node EFA example.
- OpenPI LIBERO LoRA fine-tuning.
- Isaac Lab/RSL-RL simulation training with video output.
- Multistage nut pouring pipeline adapted from the NVIDIA OSMO cookbook.
- Cosmos Reason2, Lyra, and HY-World world model examples.

### Benchmarks And Docs

- EFA/NCCL and DDP benchmark workloads for multi-node GPU validation.
- Architecture, security, reproducibility, compatibility, observability, and version matrix docs.

## Where To Start

- [Sample README](ai-infra/robotics-foundation-models-on-eks/README.md)
- [Architecture](ai-infra/robotics-foundation-models-on-eks/docs/architecture.md)
- [Examples](ai-infra/robotics-foundation-models-on-eks/examples/README.md)
- [Benchmarks](ai-infra/robotics-foundation-models-on-eks/benchmarks/README.md)
- [Observability](ai-infra/robotics-foundation-models-on-eks/docs/observability.md)

This contribution keeps runnable examples and representative `artifacts/` only. Per-example `validation.md` files are intentionally omitted; the sample run results are summarized below for PR review.

## Sample Run Results

| Area | Representative result |
| --- | --- |
| CPU smoke | `aws-osmo-smoke-9` completed through OSMO backend storage and KAI scheduling. |
| GPU smoke | `aws-osmo-gpu-smoke-2` ran on `g7e.2xlarge` with RTX PRO 6000 Blackwell, driver `580.126.09`, CUDA `13.0`. |
| Workflow patterns | `parallel-eval` completed 4-way fan-out/fan-in in `67s`; `sequential-policy` completed CPU/GPU/CPU flow in `116s` after GPU prewarm. |
| GR00T SO100 | 10k-step run completed on `g7e.8xlarge`; open-loop MSE improved from `1195.1437` to `774.3192` versus the 1-step baseline. |
| GR00T two-node EFA | `aws-gr00t-efa-multinode-osmo-smoke-cleanup-1` completed on two `g6e.8xlarge` nodes with real GR00T DDP over NCCL Libfabric/EFA; output dataset `gr00t-efa-multinode-osmo-smoke-cleanup-artifacts:1` uploaded `621.0 KiB` with `retain_model_weights=false`. |
| OpenPI LIBERO LoRA | 30k-step run completed on `g7e.4xlarge`; first-action MSE improved from `0.3599` base to `0.0015` fine-tuned. |
| Isaac Lab RSL-RL | `Isaac-Reach-Franka-v0` trained for 200 iterations on `g7e.2xlarge`; exported before/after videos and final checkpoint. |
| Nut pouring pipeline | Reproduced the six-stage NVIDIA OSMO cookbook shape through GR00T fine-tuning; final dataset `PhysAI-GR00T-Finetuned:7` retained checkpoints `3000` through `10000`; final train loss `0.0248`. |
| World models | Cosmos Reason2 NIM, HY-World 2.0 WorldMirror, and Lyra-2.0 DMD workflows completed on G7e-backed OSMO runs. |
| EFA benchmarks | G6e OSMO workflows completed on `g6e.8xlarge`: DDP EFA averaged `0.022148s/step`, Socket averaged `0.037319s/step`, and NCCL all-reduce logged `Selected provider is efa` with `Using network Libfabric`. |
| Observability | AMP/AMG path captured GPU workflow DCGM metrics through the `AWS OSMO Overview` dashboard. |

### Policy Training: GR00T SO100 Fine-Tune

GR00T SO100 1-step vs 10k action output

### Policy Training: OpenPI LIBERO LoRA

OpenPI action replay GIF

### Simulation: Isaac Lab RSL-RL Before And After

Isaac Lab before/after GIF

### Simulation: Nut Pouring Policy Rollout

Nut pouring policy rollout GIF

### World Model Inference: Cosmos Reason2 NIM

| Input | Output (is this data usable?) |
| --- | --- |
| Cosmos Reason2 input video preview | Model output from `answer.txt`:<br /><br /><strong>Decision: Reject</strong><br /><br />There is a noticeable discontinuity in the video, with abrupt changes in the scene and lack of smooth transitions between frames. The perspective shifts inconsistently, suggesting either a glitch or an editing error. Additionally, the presence of multiple vehicles and their interactions appear disjointed, lacking coherence in terms of motion and spatial relationships. The lighting and shadows also seem inconsistent, indicating potential issues with the recording or editing process. |

### World Model Inference: HY-World 2.0 WorldMirror

| Input | Output (reconstructed scene geometry) |
| --- | --- |
| HY-World Dining Table input frame | HY-World point cloud preview |

### World Model Inference: Lyra-2.0 DMD

| Input | Output (generated scene trajectory) |
| --- | --- |
| Lyra sample 04 input image<br />Prompt: `A cinematic drone shot glides forward through the center of a sun-drenched, ancient stone amphitheater...` | Lyra generated scene preview |

### Multi-Node EFA Benchmarks On G6e

G6e EFA DDP training time

G6e EFA NCCL bandwidth

### Observability: AMG GPU Utilization

AMG GPU utilization panel

## Validation

- `git ls-files 'ai-infra/README.md' 'ai-infra/robotics-foundation-models-on-eks/**/*.md' | xargs npx --yes markdownlint-cli2`
- Local markdown link check for repository-relative links.
- `git ls-files 'ai-infra/robotics-foundation-models-on-eks/**/*.sh' | xargs bash -n`
- `git ls-files 'ai-infra/robotics-foundation-models-on-eks/**/*.sh' | xargs shellcheck`
- `git ls-files 'ai-infra/robotics-foundation-models-on-eks/**/*.py' | xargs python3 -m py_compile`
- Template-token YAML parse check for the edited workflow files.
- `git diff --check`

## Contribution Confirmation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ddynwzh1992** [comment]
Screenshot 2026-05-19 at 1 13 52 pm
The EKS API endpoint access is private which does not allow further deployment from local
